pkg/controller/usersignupcleanup/usersignup_cleanup_controller.go

```diff
@@ -0,0 +1,155 @@
+	if err != nil {
+		return err
+	}
+
+	// Watch for changes to the secondary resource UserSignup and requeue the owner UserSignup
+	if err := c.Watch(&source.Kind{Type: &toolchainv1alpha1.UserSignup{}},
+		&handler.EnqueueRequestForObject{},
+	); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+var _ reconcile.Reconciler = &ReconcileUserCleanup{}
+
+// ReconcileUserCleanup cleans up old UserSignup resources
+type ReconcileUserCleanup struct {
+	client    client.Client
+	scheme    *runtime.Scheme
+	crtConfig *crtCfg.Config
+}
+
+// Reconcile reads that state of the cluster for a UserSignup object and makes changes based on the state read
+// and what is in the UserSignup.Spec
+// Note:
+// The Controller will requeue the Request to be processed again if the returned error is non-nil or
+// Result.Requeue is true, otherwise upon completion it will remove the work from the queue.
+func (r *ReconcileUserCleanup) Reconcile(request reconcile.Request) (reconcile.Result, error) {
+	reqLogger := log.WithValues("Request.Namespace", request.Namespace, "Request.Name", request.Name)
+	reqLogger.Info("Reconciling UserSignup")
+
+	// Fetch the UserSignup instance
+	instance := &toolchainv1alpha1.UserSignup{}
+	err := r.client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			// Request object not found, could have been deleted after reconcile request.
+			// Owned objects are automatically garbage collected. For additional cleanup logic use finalizers.
+			// Return and don't requeue
+			return reconcile.Result{}, nil
+		}
+		// Error reading the object - requeue the request.
+		return reconcile.Result{}, err
+	}
+	reqLogger = reqLogger.WithValues("username", instance.Spec.Username)
+
+	if instance.Spec.VerificationRequired && !instance.Spec.Approved {
+
+		createdTime := instance.ObjectMeta.CreationTimestamp
+
+		unverifiedThreshold := time.Now().Add(-time.Duration(r.crtConfig.GetUserSignupUnverifiedRetentionDays()*24) * time.Hour)
+
+		if createdTime.Time.Before(unverifiedThreshold) {
+			reqLogger.Info("Deleting UserSignup due to exceeding unverified retention period")
+			return reconcile.Result{}, r.DeleteUserSignup(instance, reqLogger)
+		}
+
+		// Requeue this for reconciliation after the time has passed between the last active time
+		// and the current unverified user deletion expiry threshold
+		requeueAfter := createdTime.Sub(unverifiedThreshold)
+
+		// Requeue the reconciler to process this resource again after the threshold for unverified user deletion
+		return reconcile.Result{
+			Requeue:      true,
+			RequeueAfter: requeueAfter,
+		}, nil
+	}
+
+	if instance.Spec.Deactivated {
+		// Find the UserSignupComplete condition
+		cond, found := condition.FindConditionByType(instance.Status.Conditions, toolchainv1alpha1.UserSignupComplete)
+
+		if !found || cond.Reason != toolchainv1alpha1.UserSignupUserDeactivatedReason {
```

Comment:
I guess we also need to check if the condition is set to true. We should not have Condition Complete set to False with Deactivated reason but just to be explicit here...

```suggestion
		if !found || cond.Status != apiv1.ConditionTrue || cond.Reason != toolchainv1alpha1.UserSignupUserDeactivatedReason {
```

with import:
```
apiv1 "k8s.io/api/core/v1"
```

Reply:
Fixed